### PR TITLE
Updated subscriber to allow mapping from custom MessageTypes to handl…

### PR DESCRIPTION
**Pull request by srpeirce**
…ers.

- This also configures rules in the subscription.

### CODE REVIEW

File: Pat.Subscriber/MessageMapping/AssemblyScanner.cs [hunk 1]
```diff
@@ -30,7 +30,15 @@ public static IEnumerable<Type> AllMessageTypesHandledBy(Type possibleHandlerTyp
...
-                .Select(handlesInterface => handlesInterface.GenericTypeArguments[0]);
+                .SelectMany(NewMethod);
+
+        private static IEnumerable<Type> NewMethod(Type handlesInterface)
+        {
+            yield return handlesInterface.GenericTypeArguments[0];
+
+
+
+        }
```
Comment (srpeirce) [inline Pat.Subscriber/MessageMapping/AssemblyScanner.cs]: Sort this...

File: TestSubscriber/TestSubscriber.csproj [hunk 2]
```diff
@@ -1,14 +1,15 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net462</TargetFramework>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
```
Comment (DaveAurionix) [inline TestSubscriber/TestSubscriber.csproj]: 2.2?

File: Pat.Subscriber.UnitTests/Pat.Subscriber.UnitTests.csproj [hunk 3]
```diff
@@ -1,6 +1,6 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net462</TargetFramework>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
```
[the remark this answers is not in the document]
Comment (srpeirce) [reply Pat.Subscriber.UnitTests/Pat.Subscriber.UnitTests.csproj]: Hmmmm no, no reason at all lol. I haven't done much dev lately, I forgot what version were on :P

File: TestSubscriber/TestSubscriber.csproj (hunk 2 above)
Comment (srpeirce) [reply TestSubscriber/TestSubscriber.csproj]: Yep doh

File: Pat.Subscriber/MessageMapping/AssemblyScanner.cs [hunk 4]
```diff
@@ -27,11 +27,18 @@ public static IEnumerable<Type> AllTypesInAssemblies(Assembly[] assembliesToScan
...
+            return possibleHandlerType
                 .GetInterfaces()
                 .Where(IsHandlerInterface)
-                .Select(handlesInterface => handlesInterface.GenericTypeArguments[0]);
-
+                .SelectMany(SelectType);
+            
+            IEnumerable<Type> SelectType(Type handlesInterface)
+            {
+                yield return handlesInterface.GenericTypeArguments[0];
```
[the remark this answers is not in the document]
Comment (srpeirce) [reply Pat.Subscriber/MessageMapping/AssemblyScanner.cs]: Cool yah, I'll review that bit. Many thanks!

**Pull request merged**